File: Python/concatenation_two_arrays.py
# https://leetcode.com/problems/concatenation-of-array/
# Concatenation of Array return [arr, arr]

def getConcatenation(nums):
    for ind in range(len(nums)):
        nums.append(nums[ind])
    return nums

# Or
def getConcatenation(nums):
    return nums + nums

# Or 
def getConcatenation(nums):
    return nums * 2


nums = [1, 2, 3]


if __name__ == '__main__':
    print(getConcatenation(nums))

File: rust/concatenation_two_arrays.rs
// https://leetcode.com/problems/concatenation-of-array/
// Concatenation of Array return [arr, arr]

fn main() {
    let array = vec![1, 2, 3];
    println!("{:?}", get_concatenation(array));

    let another_array = vec![42, 18, 10];
    println!("{:?}", get_concatenation_faster(another_array));
} 

fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
    let clone_array = nums.clone();
    [nums, clone_array].concat()
}

fn get_concatenation_faster(mut nums: Vec<i32>) -> Vec<i32> {
    let n = nums.len();
    for i in 0..n {
        nums.push(nums[i]);
    }

    nums
}